packages/rrdom/src/diff.ts

```diff
@@ -136,14 +136,27 @@ export function diff(
           break;
         }
         case 'CANVAS':
-          (newTree as RRCanvasElement).canvasMutations.forEach(
-            (canvasMutation) =>
+          {
+            const rrCanvasElement = newTree as RRCanvasElement;
+            // This canvas element is created with initial data in an iframe element. https://github.com/rrweb-io/rrweb/pull/944
+            if (rrCanvasElement.rr_dataURL !== null) {
+              const image = document.createElement('img');
+              image.src = rrCanvasElement.rr_dataURL;
+              image.onload = () => {
+                const ctx = (oldElement as HTMLCanvasElement).getContext('2d');
+                if (ctx) {
+                  ctx.drawImage(image, 0, 0, image.width, image.height);
+                }
+              };
```

Comment:
These should be flipped to insure onload isn't called before the event handler gets added
```suggestion
              image.onload = () => {
                const ctx = (oldElement as HTMLCanvasElement).getContext('2d');
                if (ctx) {
                  ctx.drawImage(image, 0, 0, image.width, image.height);
                }
              };
              image.src = rrCanvasElement.rr_dataURL;
```

Reply:
Flipped one is more straightforward for sure. But I suddenly realize that according to the JS engine's execution mechanism, I think the onload callback won't be called before the function definition. 
image.onload = ... definition is synchronous while the onload callback is asynchronous so that the callback will be always invoked after the function definition.